apps/web/src/components/StatusItem/StatusItem.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from "react";
+
+import CheckCircleIcon from "@heroicons/react/20/solid/CheckCircleIcon";
+import ExclamationCircleIcon from "@heroicons/react/20/solid/ExclamationCircleIcon";
+
+import { Link, IconType, ScrollToLink } from "@gc-digital-talent/ui";
+
+export type Status = "error" | "success";
+export type StatusColor = "default" | Status;
+
+const textColorMap: Record<StatusColor, Record<string, string>> = {
+  default: {
+    "data-h2-color": "base(black) base:hover(primary)",
+  },
+  error: {
+    "data-h2-color": "base(error.darker) base:hover(error.darkest)",
+  },
+  success: {
+    "data-h2-color": "base(success) base:hover(success.darker)",
+  },
+};
+
+const iconColorMap: Record<StatusColor, Record<string, string>> = {
+  default: {
+    "data-h2-color": "base(black) base:hover(primary)",
+  },
+  error: {
+    "data-h2-color": "base(error)",
+  },
+  success: {
+    "data-h2-color": "base(success)",
+  },
+};
+
+// could be a regular link, a scroll link, or just a regular span
+const StatusItemTitle = ({
+  href,
+  scrollTo,
+  children,
+  ...rest
+}: {
+  href?: string;
+  scrollTo?: string;
+  children?: React.ReactElement;
+}) => {
+  if (href) {
+    return (
+      <Link href={href} {...rest}>
+        {children}
+      </Link>
+    );
+  }
+  if (scrollTo) {
+    return (
+      <ScrollToLink to={scrollTo} {...rest}>
+        {children}
+      </ScrollToLink>
+    );
+  }
+  return <span {...rest}>{children}</span>;
+};
+
+export interface StatusItemProps {
+  title: string;
+  titleColor?: StatusColor;
+  status?: Status;
+  icon?: IconType;
+  iconColor?: StatusColor;
+  href?: string;
+  scrollTo?: string;
+  hiddenContextPrefix?: string;
+  asListItem?: boolean;
+  itemCount?: number;
+}
+
+export const StatusItem = ({
+  title,
+  titleColor = "default",
+  status,
+  icon,
+  iconColor = "default",
+  href,
+  scrollTo,
+  hiddenContextPrefix,
+  asListItem = true,
+  itemCount,
+}: StatusItemProps) => {
+  let Icon: IconType | null | undefined;
+  switch (status) {
+    case "error":
+      Icon = ExclamationCircleIcon;
+      break;
+    case "success":
+      Icon = CheckCircleIcon;
+      break;
+    default:
+      Icon = icon;
+  }
+
+  const Wrapper = asListItem ? "li" : "span";
+
+  // combine the context prefix for a11y if available
+  const combinedTitle = (
+    <>
+      {hiddenContextPrefix ? (
+        <span data-h2-visually-hidden="base(invisible)">
+          {`${hiddenContextPrefix} - `}
+        </span>
+      ) : null}
+      {title}
+    </>
+  );
+
+  const effectiveIconColor = status ?? iconColor;
+  const effectiveTitleColor = status === "error" ? "error" : titleColor;
+
+  return (
+    <Wrapper
+      data-h2-display="base(flex)"
+      data-h2-flex-direction="base(row)"
```

Comment:
If you don't feel like being explicit, we could remove this line since row is the default direction for flex.

Reply:
:+1: [remove redundant attribute](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6823/commits/1f7a8d2c6ef34bdf20e4477ba4b6357b8fdcf43a)

Reply:
It's 👍 all the way down!